File: src/main/java/io/scriptor/rest/RootRest.java
package io.scriptor.rest;

import io.scriptor.annotation.Endpoint;
import io.scriptor.annotation.Resource;

import java.io.InputStream;

@Endpoint("/")
public class RootRest {

    @Resource(path = "favicon.[]", result = "image/svg+xml")
    public InputStream getFavicon() {
        return ClassLoader.getSystemResourceAsStream("favicon.svg");
    }
}

File: src/main/java/io/scriptor/rest/RootRest.kt
package io.scriptor.rest

import io.scriptor.annotation.Endpoint
import io.scriptor.annotation.Resource
import java.io.InputStream

@Endpoint("/")
class RootRest {

    @Resource(path = "favicon.[]", result = "image/svg+xml")
    fun getFavicon(): InputStream? = ClassLoader.getSystemResourceAsStream("favicon.svg")
}
